docs/conf.py

```diff
@@ -127,6 +157,48 @@
 htmlhelp_basename = 'LightGBMdoc'
 
 
+def generate_doxygen_xml(app):
+    """Generate XML documentation for C API by Doxygen.
+
+    Parameters
+    ----------
+    app : object
+        The application object representing the Sphinx process.
+    """
+    doxygen_args = [
+        "INPUT={}".format(os.path.join(CURR_PATH, os.path.pardir,
+                                       'include', 'LightGBM', 'c_api.h')),
+        "OUTPUT_DIRECTORY={}".format(os.path.join(CURR_PATH, 'doxyoutput')),
+        "GENERATE_HTML=NO",
+        "GENERATE_LATEX=NO",
+        "GENERATE_XML=YES",
+        "XML_OUTPUT=xml",
+        "XML_PROGRAMLISTING=YES",
+        r'ALIASES="rst=\verbatim embed:rst:leading-asterisk"',
+        r'ALIASES+="endrst=\endverbatim"',
+        "ENABLE_PREPROCESSING=YES",
+        "MACRO_EXPANSION=YES",
+        "EXPAND_ONLY_PREDEF=NO",
+        "SKIP_FUNCTION_MACROS=NO",
+        "SORT_BRIEF_DOCS=YES",
+        "WARN_AS_ERROR=YES",
+    ]
+    doxygen_input = '\n'.join(doxygen_args)
+    if sys.version[0] == "3":
+        doxygen_input = bytes(doxygen_input, "utf-8")
+    if not os.path.exists(os.path.join(CURR_PATH, 'doxyoutput')):
+        os.makedirs(os.path.join(CURR_PATH, 'doxyoutput'))
+    try:
+        with open(os.devnull, "w") as devnull_file:
+            process = Popen(["doxygen", "-"],
+                            stdin=PIPE, stdout=devnull_file, stderr=STDOUT)
```

Comment:
Why need to shut up Doxygen:
https://github.com/svenevs/exhale/blob/fe7644829057af622e467bb529db6c03a830da99/exhale/deploy.py#L99-L111

Reply:
I'm open to any suggestions! Maybe redirect output to a log file and print it in case of failure?..

Reply:
@henry0312 Can you please share your opinion about this?

Reply:
I think

```python
stdin=PIPE, stdout=subprocess.DEVNULL, stderr=None)
```

is better, but why do you want to shut up Doxygen outputs (`stdout=devnull_file`) ?

Reply:
Ah, is that a problem of RTD?

Reply:
@henry0312 
`subprocess.DEVNULL` is not available in 2.7.
https://docs.python.org/3/library/subprocess.html#subprocess.DEVNULL

Please read:
> Why need to shut up Doxygen:
https://github.com/svenevs/exhale/blob/fe7644829057af622e467bb529db6c03a830da99/exhale/deploy.py#L99-L111

Reply:
> `subprocess.DEVNULL` is not available in 2.7.

Oh, I see.

Will we really suffer from the problem?

Reply:
@henry0312 I have no idea! 😄  I just found that comment and wanted to ask your opinion/experience...

Reply:
I don't know, too.
If the problem doesn't happen, I don't think the hack won't be needed.

Reply:
@henry0312 Fair point! So, I'll remove it and leave a comment about that this can cause RTD silent failure. OK?